x/auction/types/genesis.go

```diff
@@ -4,51 +4,80 @@ import (
 	"bytes"
 	"fmt"
 
+	types "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	proto "github.com/gogo/protobuf/proto"
 )
 
-// DefaultNextAuctionID is the starting poiint for auction IDs.
+// DefaultNextAuctionID is the starting point for auction IDs.
 const DefaultNextAuctionID uint64 = 1
 
-// GenesisAuction is an interface that extends the auction interface to add functionality needed for initializing auctions from genesis.
+// GenesisAuction extends the auction interface to add functionality
+// needed for initializing auctions from genesis.
 type GenesisAuction interface {
 	Auction
 	GetModuleAccountCoins() sdk.Coins
 	Validate() error
 }
 
-// GenesisAuctions is a slice of genesis auctions.
-type GenesisAuctions []GenesisAuction
+// PackGenesisAuctions converts a GenesisAuction slice to Any slice
+func PackGenesisAuctions(ga []GenesisAuction) ([]*types.Any, error) {
+	gaAny := make([]*types.Any, len(ga))
+	for i, genesisAuction := range ga {
+		msg, ok := genesisAuction.(proto.Message)
+		if !ok {
+			return nil, fmt.Errorf("cannot proto marshal %T", genesisAuction)
+		}
+		any, err := types.NewAnyWithValue(msg)
+		if err != nil {
+			return nil, err
+		}
+		gaAny[i] = any
+	}
+
+	return gaAny, nil
+}
+
+// UnpackGenesisAuctions converts Any slice to GenesisAuctions slice
+func UnpackGenesisAuctions(genesisAuctionsAny []*types.Any) ([]GenesisAuction, error) {
+	genesisAuctions := make([]GenesisAuction, len(genesisAuctionsAny))
+	for i, any := range genesisAuctionsAny {
+		genesisAuction, ok := any.GetCachedValue().(GenesisAuction)
+		if !ok {
+			return nil, fmt.Errorf("expected genesis auction")
+		}
+		genesisAuctions[i] = genesisAuction
+	}
 
-// GenesisState is auction state that must be provided at chain genesis.
-type GenesisState struct {
-	NextAuctionID uint64          `json:"next_auction_id" yaml:"next_auction_id"`
-	Params        Params          `json:"params" yaml:"params"`
-	Auctions      GenesisAuctions `json:"auctions" yaml:"auctions"`
+	return genesisAuctions, nil
 }
 
 // NewGenesisState returns a new genesis state object for auctions module.
-func NewGenesisState(nextID uint64, ap Params, ga GenesisAuctions) GenesisState {
-	return GenesisState{
-		NextAuctionID: nextID,
+func NewGenesisState(nextID uint64, ap Params, ga []GenesisAuction) *GenesisState {
+	packedGA, err := PackGenesisAuctions(ga)
+	if err != nil {
+		panic(err)
+	}
+	return &GenesisState{
+		NextAuctionId: nextID,
 		Params:        ap,
-		Auctions:      ga,
+		Auctions:      packedGA,
 	}
 }
 
 // DefaultGenesisState returns the default genesis state for auction module.
-func DefaultGenesisState() GenesisState {
+func DefaultGenesisState() *GenesisState {
 	return NewGenesisState(
 		DefaultNextAuctionID,
 		DefaultParams(),
-		GenesisAuctions{},
+		[]GenesisAuction{},
 	)
 }
 
 // Equal checks whether two GenesisState structs are equivalent.
 func (gs GenesisState) Equal(gs2 GenesisState) bool {
```

Comment:
i believe we can remove all genesis Equals func since its not used and avoids using amino

Reply:
https://github.com/Kava-Labs/kava/pull/1052/commits/f87207113997c5f58e3483e396fbfcfd31515ce2